src/main/java/com/puppycrawl/tools/checkstyle/utils/CodePointUtil.java

```diff
@@ -42,4 +42,18 @@ public static boolean isBlank(int... codePoints) {
                 .allMatch(Character::isWhitespace);
     }
 
+    /**
+     * Checks if the given code point array contains only whitespace up to specified index.
+     *
+     * @param codePoints
+     *            array of Unicode code point of string to check
+     * @param index
+     *            index to check up to (exclusive)
+     * @return true if all code points preceding given index are whitespace
+     */
+    public static boolean hasWhitespaceBefore(int index, int... codePoints) {
+        return Arrays.stream(codePoints, 0, index)
+                .allMatch(Character::isWhitespace);
```

Comment:
by the way, now above method `isBlank(int... codePoints)` can be simplified to 
```
public static boolean isBlank(int... codePoints) {
        return hasWhitespacesBefore(codePoints.length, codePoints);
    }
```

Reply:
absolutely right, done